### ISSUE TITLE
How to install server-side libreoffice on Ubuntu?

### ISSUE THREAD

**Issue by binarykitchen** [owncloud/documents#478]
To enable `MS Word support (requires openOffice/libreOffice)` I will have to install libreOffice but the default `apt-get install libreoffice` installs way too much, including Java and so on.

Don't want to install Java on the server side. It's too heavy and consumes too many resources. Isn't there a slim libreoffice installation for that?

Or is there another way around?

**Comment by RealRancor** [comment]
Hi,

as this is no bug in the app please ask such questions either at the forums, irc channel or user mailinglist.

**Comment by VicDeo** [comment]
@binarykitchen libreoffice is written in Java so there is a little sense to install libreoffice only. It just won't work.

**Comment by RealRancor** [comment]
Yes, thats also noted here: https://www.libreoffice.org/get-help/system-requirements/

> For certain features of the software - but not most - Java is required. Java is notably required for Base.

**Comment by binarykitchen** [comment]
hmmm, that sucks. aren't there any other non-java alternatives to enable document rendering on the server side?

**Comment by kossebau** [comment]
LibreOffice is written more in C++ than Java ;) Note also the "but not most" in  "For certain features of the software - but not most - Java is required. Java is notably required for Base."
Then I have no idea if Java is required for anything needed by thumbnailing or document conversion, best ask the LibreOffice people for first-hand info: https://www.libreoffice.org/get-help/community-support/
Another problem is if LO packagers of your favourite linux version made it possible to install Java-free subset of LO...

> aren't there any other non-java alternatives to enable document rendering on the server side?

Calligra would be a candidate there, no Java required and, being improved e.g. by Nokia for the document viewer on N900 and N9, support for MS formats and ODF is quite good. But then we have the same problem of packagers, given office suites being a monster to package and they forward the hell usually to users by throwing everything in a few packages and make anything hard dependencies. Noone to blame, it is really complicated (tells you the person who tried to make Calligra internal deps managable at least for developers :) ).

**Comment by kossebau** [comment]
PS: There is hope later this year for a more slim Calligra, given the ongoing Calligra port to Qt5 and KF5, where reducing the footprint of deps is a high priority.

**Comment by binarykitchen** [comment]
okay, thanks for the input. i guess i better wait for when Calligra is ready. libreoffice is too heavy for me.

**Comment by ram4nd** [comment]
Got a java warning:
```
javaldx: Could not find a Java Runtime Environment!
Warning: failed to read path from javaldx
```
But the command ran through and did generate the csv for me.
